prov/efa/src/rxr/rxr.h

```diff
@@ -72,13 +72,20 @@
 
 
 #ifdef ENABLE_EFA_POISONING
-extern const uint32_t rxr_poison_value;
-static inline void rxr_poison_mem_region(uint32_t *ptr, size_t size)
+static inline void rxr_poison_mem_region(void *ptr, size_t size)
 {
-	int i;
+	uint32_t rxr_poison_value = 0xdeadbeef;
+	memset(ptr, rxr_poison_value, size);
```

Comment:
I do not think `memset` does what we want. `memset` use only 1 byte of value. so the poisoned buffer will be `0xefefef` instead of `0xdeadbeef`

Reply:
you're right. I added back the loop.